src/octree/mod.rs

```diff
@@ -80,18 +80,28 @@ fn relative_size_on_screen(bounding_cube: &Cube, matrix: &Matrix4<f32>) -> f32 {
     (rv.max().x - rv.min().x) * (rv.max().y - rv.min().y)
 }
 
-#[derive(Debug)]
-pub struct OnDiskOctree {
+pub trait OctreeDataProvider: Send + Sync {
+    fn meta_proto(&self) -> Result<proto::Meta>;
+    fn data(
+        &self,
+        node_id: &NodeId,
+        node_extensions: Vec<&str>,
+    ) -> Result<HashMap<String, Box<dyn Read>>>;
+    fn number_of_points(&self, node_id: &NodeId) -> Result<i64>;
+}
+
+pub struct Octree {
```

Comment:
line 93 mod.rs: Is it enough that the trait OctreeDataProvider is Send + Sync? 
See line 75 in pointswebviewer.rs (client of web viewer)

Reply:
question already answered